cmd/keymaster/main.go

```diff
@@ -239,8 +239,10 @@ func doU2FAuthenticate(client *http.Client, authCookies []*http.Cookie, baseURL
 	}
 	logger.Debugf(0, "Get url request did not failed %+v", signRequestResp)
 
-	defer signRequestResp.Body.Close()
+	// Dont defer the body response Close ... as we need to close it explicitly
+	// in the body of the function so that we can reuse the connection
```

Comment:
Why not use deferred function which drains and closes?

Reply:
Only the last response on a function can use the defer trick, otherwise the connection cannot be reused (which is the objective of the pull request). If you notice, this function does use the defer, but only on the last. Also, the defered close does not guarantee to drain the connection.

Reply:
Maybe I'm missing something. I thought that in order to re-use the connection, you need to always drain and close the body? My suggestion would ensure that. This is what I'm suggesting:
    defer func() {
        io.Copy(ioutil.Discard, signRequestResp.Body)
       signRequestResp.Body.Close()
    }()

Reply:
Not exactly, the body must be drained and closed before doing any further requests. Your  proposed defered function would work if this function only made a single request. However since I make 2 requests on this function(doU2FAuthenticate)  I need to explicitly drain and close the first connection  in order to reuse this connection on the second request.  That what the comment was all about.

Reply:
Ah, OK. Merging. Travis CI has passed, it's just not showing on the PR page.